epp-client/src/tests/de.rs

```diff
@@ -655,6 +658,27 @@ mod response {
 
         let ext = object.extension.unwrap();
 
-        assert_eq!(ext.data.subproduct, "com".into());
+        let ext = match ext.data {
+            ExtensionResponse::NameStore(data) => data,
+            ExtensionResponse::Rgp(_) => panic!("Rgp extension found"),
+        };
+
+        assert_eq!(ext.subproduct, "com".into());
+    }
+
+    #[test]
+    fn namestore_rgp() {
+        let xml = get_xml("response/extensions/domain_info_rgp.xml").unwrap();
+
+        let object = DomainInfo::<NameStore>::deserialize_response(xml.as_str()).unwrap();
+
+        let ext = object.extension.unwrap();
+
+        let ext = match ext.data {
+            ExtensionResponse::NameStore(_) => panic!("NameStore extension found"),
```

Comment:
Maybe use `unreachable!()` here rather than the custom `panic!()`?

(Pretty sure you don't need to import `panic!()` anyway...)

Reply:
Yeah, I think my IDE sucked that in.

Is it really unreachable though? We know this is the case because we wrote the xml, but I'm not sure that's really the intent of the unreachable macro?

```
Indicates unreachable code.

This is useful any time that the compiler can't determine that some code is unreachable. For example:

- Match arms with guard conditions.
- Loops that dynamically terminate.
- Iterators that dynamically terminate.

If the determination that the code is unreachable proves incorrect, the program immediately terminates with a panic!.
```

Reply:
Hmm, I think this fits the bill for "we know statically that something crazy would have to happen to hit this case, but the compiler can't tell".